lib/clAdapters/google/errors.ts

```diff
@@ -0,0 +1,46 @@
+// Type union instead of enum so calling code can get name without type details
+export type GoogleErrorType =
+  'InvalidGrant' |
+  'UnauthorizedClient'
+;
+
+export interface InvalidGrant {
+  kind: 'InvalidGrant';
+  err: Error;
+}
+
+export interface UnauthorizedClient {
+  kind: 'UnauthorizedClient';
+  err: Error;
+}
+
+export type GoogleError = InvalidGrant | UnauthorizedClient;
+
+export function createGoogleError<T extends GoogleErrorType>(
+  kind: T,
+  err?: Error
+): GoogleError {
+  if (!err) {
+    err = new Error();
+    Error.captureStackTrace(err, createGoogleError);
+  }
+
+  // return {
+  //     kind,
+  //     err
+  // };
+  // Type value is not enforceable:
```

Comment:
Ah, good stuff. Typescript seems pretty smart w/string literal unions. I assumed the check would exhibit same behavior as enums, which don't enforce a valid value. [Example playground](http://www.typescriptlang.org/play/index.html#src=enum%20GoogleErrorType%20%7B%0D%0A%20%20%20%20InvalidGrant%20%3D%201%2C%0D%0A%20%20%20%20UnauthorizedClient%20%3D%202%0D%0A%7D%0D%0A%0D%0Atype%20GoogleError%0D%0A%20%20%3D%20InvalidGrant%0D%0A%20%20%7C%20UnauthorizedClient%3B%0D%0A%0D%0A%0D%0Ainterface%20InvalidGrant%20%7B%0D%0A%20%20%20%20kind%3A%20GoogleErrorType.InvalidGrant%3B%0D%0A%20%20%20%20err%3A%20Error%3B%0D%0A%7D%0D%0A%0D%0Ainterface%20UnauthorizedClient%20%7B%0D%0A%20%20%20%20kind%3A%20GoogleErrorType.UnauthorizedClient%3B%0D%0A%20%20%20%20err%3A%20Error%3B%0D%0A%7D%0D%0A%0D%0A%0D%0A%2F**%0D%0A%20%20*%20%20creating%20errors%0D%0A%20%20*%2F%0D%0Afunction%20createError%3CT%20extends%20GoogleErrorType%3E(%0D%0A%20%20kind%3A%20T%2C%0D%0A%20%20err%3A%20Error%20%3D%20new%20Error()%0D%0A)%20%7B%0D%0A%20%20return%20%7B%20kind%2C%20err%20%7D%3B%0D%0A%7D%0D%0A%0D%0Aconst%20test%3A%20GoogleError%20%3D%20createError(10%2C%20new%20Error('test'))%3B). I'll make the change.